tests/src/test/scala/spire/laws/cooperative_equality/AlgebraicCoopEqProp.scala

```diff
@@ -0,0 +1,52 @@
+package spire.laws.cooperative_equality
+
+import org.scalacheck.{Prop, Properties, Arbitrary, Gen}
+import org.scalacheck.Prop.forAll
+import org.scalacheck.Prop._
+import org.scalacheck.Arbitrary._
+import org.scalacheck.Test.Parameters
+
+import spire.math.{Algebraic, Rational, SafeLong}
+import spire.laws.cooperative_equality.Utility.arbitrarySafeLong
+
+object AlgebraicCoopEqProp extends Properties("Algebraic") {
+
+	val genAlgebraicInt: Gen[Algebraic]                 = arbitrary[Int].map(Algebraic(_))
+  val genAlgebraicLong: Gen[Algebraic]                = arbitrary[Long].map(Algebraic(_))
+  val genAlgebraicDouble: Gen[Algebraic]              = arbitrary[Double].map(Algebraic(_))
+  val genAlgebraicBigInt: Gen[Algebraic]              = arbitrary[BigInt].map(Algebraic(_))
+  val genAlgebraicRational: Gen[Algebraic]            = for {
+    num <- arbitrary[SafeLong]
+    neg <- Gen.choose(Long.MinValue, -1)
+    pos <- Gen.choose(1, Long.MaxValue)
+    den <- Gen.oneOf(neg, pos)
+  } yield Rational(num, den)
```

Comment:
I think you probably should be wrapping this in an `Algebraic(...)` right? Since you want to return an `Algebraic` value?